### PR TITLE
fix: [Wasm] Don't use Assemly.Load in ApiInformation

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): Fixes https://github.com/unoplatform/uno/issues/3801

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

This change avoids relying on APIs performing stack crawling on Wasm AOT.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by jeromelaban** [comment]
@mergifyio release/stable/3.0

**Comment by jeromelaban** [comment]
@Mergifyio backport release/stable/3.0

File: src/Uno.UI/UI/Xaml/Application.cs [hunk 1]
```diff
@@ -43,6 +43,12 @@ public partial class Application
 		private ApplicationTheme? _requestedTheme;
 		private bool _themeSetExplicitly = false;
 
+		static Application()
+		{
+			ApiInformation.RegisterAssembly(typeof(Application).Assembly);
+			ApiInformation.RegisterAssembly(typeof(Windows.Storage.ApplicationData).Assembly);
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Application.cs:49]: There were 3 assemblies before, there seem to be 2 registered now?

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Application.cs:49]: The other one is registered directly ApiInformation (it's itself)

**Pull request merged**